kinesis/kinesis.go

```diff
@@ -562,15 +562,32 @@ func (outputPlugin *OutputPlugin) randomString() string {
 	return string(outputPlugin.random.buffer)
 }
 
+func getFromMap(dataKey string, records map[interface{}]interface{}) interface{} {
+	for k, v := range records {
+		currentKey := stringOrByteArray(k)
+		if currentKey == dataKey {
+			return v
+		}
+	}
+
+	return nil
+}
+
 // getPartitionKey returns the value for a given valid key
 // if the given key is empty or invalid, it returns a random string
 func (outputPlugin *OutputPlugin) getPartitionKey(record map[interface{}]interface{}) string {
 	partitionKey := outputPlugin.partitionKey
+	partitionKeys := strings.Split(partitionKey, "->")
+	num := len(partitionKeys)
 	if partitionKey != "" {
-		for k, v := range record {
-			dataKey := stringOrByteArray(k)
-			if dataKey == partitionKey {
-				value := stringOrByteArray(v)
+		for count, dataKey := range partitionKeys {
+			newRecord := getFromMap(dataKey, record)
+			_, ok := newRecord.(map[interface{}]interface{})
+			if ok {
+				record = newRecord.(map[interface{}]interface{})
+			} 
+			if count == num-1 {
+				value := stringOrByteArray(newRecord)
 				if value != "" {
 					if len(value) > partitionKeyMaxLength {
```

Comment:
I think the `if count == num - 1` should be at the top of the loop... that way you don't try to cast it to a map if its not needed

Reply:
yea. It is a good idea